tests/test_util.py

```diff
@@ -281,3 +252,82 @@ def test_deep_merge() -> None:
     }
     Utility.deep_merge(x, y)
     assert x == expected
+
+
+def test_deep_merge_dict_w_list() -> None:
+    a = {
+        "a": {"i": [1, 2], "j": {"c": 3, "d": 4}},
+        "b": {"i": {"c": 5, "d": 6}, "j": {"c": 7, "d": [8, {"x": [9, 10], "y": 11}, 12]}},
+    }
+
+    b = {
+        "a": {"i": [11, 12, 13]},
+        "b": {"i": {"c": 15}, "j": {"d": [8, {"x": [19, 110]}]}},
+    }
+
+    expected = {
+        "a": {"i": [11, 12, 13], "j": {"c": 3, "d": 4}},
+        "b": {"i": {"c": 15, "d": 6}, "j": {"c": 7, "d": [8, {"x": [19, 110], "y": 11}, 12]}},
+    }

```

Comment:
```suggestion
    a = {
        "a": {"i": [1, 2, 3, 4, 5], "j": {"c": 3, "d": 4}},
        "b": {"i": {"c": 5, "d": 6}, "j": {"c": 7, "d": [8, {"x": [9, 10], "y": 11}, 12]}},
    }

    b = {
        "a": {"i": [11, 12, 13]},
        "b": {"i": {"c": 15}, "j": {"d": [8, {"x": [19, 110]}]}},
    }

    expected = {
        "a": {"i": [11, 12, 13, 4, 5], "j": {"c": 3, "d": 4}},
        "b": {"i": {"c": 15, "d": 6}, "j": {"c": 7, "d": [8, {"x": [19, 110], "y": 11}, 12]}},
    }

```
Just a thought that this could help explain how the function handles lists.

Reply:
b.j.d already covers the case where update is shorter than target. Let's keep a.i as is to cover the case where target is shorter than update.